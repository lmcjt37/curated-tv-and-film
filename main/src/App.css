.main-header {
  margin-top: 30px;
  margin-bottom: 30px;
}

.main-header h1 {
  margin: 0;
}

.content-filter {
  padding: 0 28px;
}

.content-filter input {
  font-size: 17px;
  padding: 20px 10px;
}

.content-list {
  margin-top: 60px;
  margin-bottom: 60px;
}
.form-control {
  border: 1px solid #b5bac5;
  color: unset;
}
.content-list ul {
  list-style: none;
  margin: 0;
  padding: 0;
}
ul li {
  list-style: none;
}
.content-list ul li {
  box-shadow: 0 2px 5px 0 rgba(224, 227, 231, 0.7);
  background: #fff;
  padding: 2rem;
}

.content-list ul li + li {
  margin-top: 60px;
}
.row {
  margin: 0;
  padding: 2rem;
}
.item-thumbnail {
  max-width: 30%;
  min-width: 175px;
}
.item-thumbnail img {
  max-width: 100%;
}
.item-info {
  max-width: 70%;
  flex: 1;
}
.item-imdb {
  display: block;
  margin-top: 10px;
}

.item-imdb span,
.item-link {
  font-style: italic;
  margin-right: 5px;
}

.details__item + .details__item {
  margin-top: 5px;
}

.details__item span {
  font-weight: bold;
}

.mainContainer {
  height: 100%;
  overflow: hidden;
}

.headerContainer {
  align-items: center;
  background: #fff;
  box-shadow: 0px 4px 15px rgba(100, 110, 130, 0.1);
  display: grid;
  grid-template-columns: auto 1fr auto;
  z-index: 1;
}

.logoContainer {
  display: grid;
  width: 100%;
  height: 100%;
  z-index: 1;
  padding: 0 28px;
}

.logoContainerShadow {
  box-shadow: 0px 4px 15px rgba(100, 110, 130, 0.1);
}

.logoImage {
  align-self: center;
  justify-self: left;
  max-width: 100%;
  max-height: 50px;
  font-size: 28px;
  color: #00daad;
}

.searchContainer {
  padding: 0 36px;
  position: relative;
}

.topMenuContainer {
  align-items: center;
  display: grid;
  grid-gap: 24px;
  grid-template-columns: min-content min-content;
  justify-self: right;
  padding: 0 36px;
}

.menuItem {
  justify-self: end;
}

.imageWaraper {
  display: grid;
}
.imageWaraper svg {
  align-self: center;
  justify-self: center;
  cursor: pointer;
  border-radius: 50%;
}

.item-image {
  max-width: 100%;
  margin: 0 auto;
}

.item-link {
  display: inline-block;
  margin: 0;
}

/* Footer */
footer#footer {
  width: 100%;
}

footer#footer .container {
  border-top: 1px solid #CCC;
  padding: 3rem 0 2rem;
}

footer#footer .links {
  display: flex;
  justify-content: center;
}

footer#footer .links li {
  padding: 0 1rem;
}

footer#footer .links li a {
  font-weight: bold;
  color: rgb(var(--dark-persian-blue));
}

footer#footer .links img {
  width: 24px;
  height: 24px;
}

footer#footer .links span {
  margin-left: .5rem;
}

footer#footer .version {
  text-align: center;
}

@media (max-width: 767px) {
  .item-image {
    min-width: 100%;
    margin-bottom: 1rem;
  }

  .item-thumbnail {
    max-width: 100%;
  }

  .item-info {
    max-width: none;
    padding: 0;
  }

  .logoImage {
    font-size: 24px;
  }

  .headerContainer {
    display: block;
    padding-top: 10px;
    padding-bottom: 20px;
  }

  .content-list ul li + li {
    margin-top: 30px;
  }
}

@media (min-width: 768px) {
  .item-card {
    display: flex;
  }

  .item-thumbnail {
    margin-right: 30px;
  }

  .item-image {
    width: 266px;
    height: auto;
  }

  .headerContainer {
    height: 60px;
  }
}
