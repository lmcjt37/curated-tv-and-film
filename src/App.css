body {
  font-family: 'Roboto', sans-serif;
}

ul {
  list-style-type: none;
  padding: 0;
}
