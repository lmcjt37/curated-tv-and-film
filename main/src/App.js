import React, { Component } from 'react';

import './App.css';
import content from './content.js';
import icons from './assets/icons';

var data = content;

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchString: "",
      content: content,
      filterTV: true,
      filterMovies: true,
      filterYear: ""
    }
  }

  handleChange = (event) => {
    var searchString = event.target.value;
    this.setState({searchString: searchString});

    searchString = searchString.trim().toLowerCase();

    if(searchString.length > 0) {
      var searchResult = this.state.content.filter((el) =>
        el.title.toLowerCase().match(searchString));
      this.setState({content: searchResult});
    } else {
      this.setState({content: data});
    }
  }

  handleYear = (event) => {
    let year = event.target.value;
    this.setState({filterYear: year});
  }

  handleFilter = (event) => {
    this.setState({ [event.target.name] : event.target.checked});
    this.setState({filterYear: ""});
  }

  render() {
    return (
      <div className="mainContainer">
          <div className="headerContainer">
          <div className="logoContainer">
            <div className="logoImage">Curated TV and Film</div>
          </div>
          <div className="content-filter">
            <input type="text" className="form-control" value={this.state.searchString} onChange={this.handleChange} placeholder="Search here"/>
          </div>
          <div className='filter-div'>
              <input type="checkbox" name = "filterMovies" checked={this.state.filterMovies} onChange={this.handleFilter}/><label htmlFor='movies' className="filter-labels">Movies</label>
              <input type="checkbox" name = "filterTV" checked={this.state.filterTV} onChange={this.handleFilter}/><label htmlFor='tv' className="filter-labels">TV Series</label>
              {this.state.filterMovies && !this.state.filterTV ? (
                  <input type="text" name = "filterYear" value={this.state.filterYear} onChange={this.handleYear} placeholder="Year"/>
                ) : (
                  null
                )
              }
          </div>
        </div>
      <section className="app container">
        <main className="main-content">
          <div className="content-list">
            {this.state.content.length ? (
              <ul>
                {this.state.content.map((item, index) => {
                  if (this.state.filterMovies === false && item.type === "movie") 
                      return "";
                  if (this.state.filterTV === false && item.type === "tv_show" )  
                     return "";
                  if (this.state.filterYear != "" && this.state.filterYear != item.year)
                     return "";
                  return (
                    <li key={index}>
                      <div className="item-card">
                        <div className="item-thumbnail">
                          <a target="_blank"  rel="noopener noreferrer" href={item.url}>
                            <img src={item.thumbnail} alt={item.title} className="item-image"/>
                          </a>
                        </div>
                        <div className="item-info">
                          <h2>{item.title}{item.year ? (" (" + item.year + ")") : ''}</h2>
                          {item.type === "tv_show" &&
                            <div className="tvshow-details">
                              <div className="details__item"><span>Season:</span> {item.season}</div>
                              <div className="details__item"><span>Episode:</span> {item.episode}</div>
                              <div className="details__item"><span>Episode Title:</span> {item.episode_title}</div>
                            </div>
                          }
                          <div className="item-desc details__item"><span>Description:</span> {item.description}</div>
                          <div className="item-imdb">
                            <span>IMDB:</span><a target="_blank"  rel="noopener noreferrer" href={item.imdb} className="item-link">{item.imdb}</a>
                          </div>
                        </div>
                      </div>
                    </li>
                  )
                })}
              </ul>
              ) : (
              this.state.searchString ? (
                <p>No search result</p>
                ) : (
                <p>Can't load the data.</p>
                )
              )
            }
          </div>
        </main>
      </section>
      <footer id="footer">
        <div className="container">
          <ul className="links">
            <li>
              <a href="https://github.com/lmcjt37/curated-tv-and-film">
                <img src={icons.github} alt="Repository" />
                <span>Repository</span>
              </a>
            </li>
            <li>
              <a href="https://github.com/lmcjt37/curated-tv-and-film/graphs/contributors">
                <img src={icons.github} alt="Contributors" />
                <span>Contributors</span>
              </a>
            </li>
          </ul>
          <p class="version">v0.0.1</p>
        </div>
      </footer>
      </div>
    );
  }
}

export default App;
