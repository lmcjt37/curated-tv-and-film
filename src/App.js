// React
import React, { useState, useEffect } from 'react';
import { Events, animateScroll as scroll } from 'react-scroll';

// Components
import './App.css';
import data from './content.js';
import Header from './components/header.js';
import Footer from './components/footer.js';
import FilterBar from './components/filterBar.js';
import Card from './components/card.js';
import MultiCard from './components/multiCard';
import Tile from './components/tile';

// Material UI - Core
import Grid from '@material-ui/core/Grid';

// Material UI
import ErrorIcon from '@material-ui/icons/Error';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  content: {
    marginTop: `${theme.spacing(2) + 54}px`,
    [theme.breakpoints.up('sm')]: {
      marginTop: `${theme.spacing(2) + 64}px`
    }
  },
  error: {
    textAlign: 'center',
    width: '100%',
    margin: '200px auto'
  }
});

// helper functions
export const handleYear = (event, onlyMovies, setFilterYear, setContent) => {
  let year = event.target.value;
  let moviesFilteredByYear = onlyMovies;
  if (year !== 'All') {
    moviesFilteredByYear = onlyMovies.filter(
      movie => movie.year.toString() === year.toString()
    );
  }
  setFilterYear(year);
  setContent(moviesFilteredByYear);
};

export const handleChange = (
  event,
  data,
  setAutoComplete,
  setSearch,
  setContent
) => {
  let searchString = event.target.value.trim().toLowerCase();

  if (searchString) {
    let searchResult = data.filter(
      el =>
        el.title.toLowerCase().match(searchString) ||
        (el.year && el.year.toString().match(searchString))
    );

    setAutoComplete(
      data
        .map(el =>
          el.title.toLowerCase().match(searchString)
            ? { label: el.title }
            : null
        )
        .filter(item => item !== null)
    );

    setSearch(true);
    setContent(searchResult);
  } else {
    setSearch(false);
    setContent(data);
    setAutoComplete([]);
  }
};

export const handleFilter = (
  event,
  data,
  content,
  setFilterResults,
  setContent,
  setYears,
  setFilterYear,
  setOnlyMovies
) => {
  setFilterResults(event.target.value);
  setContent(data);
  setYears([]);
  setFilterYear('All');

  if (event.target.value !== 'tv') {
    let filteredMovies = content.filter(value => value.type === 'movie');
    let onlyYears = new Set(filteredMovies.map(movie => movie.year));
    setYears([...onlyYears].sort((a, b) => b - a));
    setOnlyMovies(filteredMovies);
  }
};

export const handleToggleChip = (
  chip,
  filterGenre,
  setFilterGenre,
  setShowFilters
) => {
  let _filterGenre = filterGenre;

  let index = _filterGenre.on.indexOf(chip);
  if (index > -1) {
    _filterGenre.on.splice(index, 1);
  } else {
    _filterGenre.on.push(chip);
  }
  setFilterGenre({ ..._filterGenre });
  setShowFilters(true);
};

export const handleOrder = (event, setFilterOrder) => {
  setFilterOrder(event.target.value);
};

export const goTop = () => {
  scroll.scrollToTop();
};

const App = ({ classes, testing = false, testType = null }) => {
  const [search, setSearch] = useState(false);
  const [content, setContent] = useState(data);
  const [showFilters, setShowFilters] = useState(false);
  const [showGrid, setShowGrid] = useState(false);
  const [filterResults, setFilterResults] = useState('all');
  const [filterYear, setFilterYear] = useState('All');
  const [filterGenre, setFilterGenre] = useState({
    available: [
      'Action',
      'Adventure',
      'Animation',
      'Biography',
      'Comedy',
      'Crime',
      'Drama',
      'Family',
      'Fantasy',
      'History',
      'Horror',
      'Mystery',
      'Romance',
      'Sci-Fi',
      'Thriller'
    ],
    on: []
  });
  const [filterOrder, setFilterOrder] = useState('Ascending');
  const [onlyMovies, setOnlyMovies] = useState([]);
  const [years, setYears] = useState([]);
  const [autoComplete, setAutoComplete] = useState([]);

  useEffect(() => {
    //testing code
    if (testing === true) {
      if (testType === 'no data') {
        setContent([]); // testing - 'can't load the data'
      }
      if (testType === 'handleYear') {
        setShowFilters(true);
        setFilterResults('movies');
      }
    }

    //end of testing code
    Events.scrollEvent.register('begin', () => {});
    Events.scrollEvent.register('end', () => {});
  }, [testing, testType]);

  const callHandleChange = event => {
    handleChange(event, data, setAutoComplete, setSearch, setContent);
  };

  const callHandleYear = event => {
    handleYear(event, onlyMovies, setFilterYear, setContent);
  };

  const callHandleFilter = event => {
    handleFilter(
      event,
      data,
      content,
      setFilterResults,
      setContent,
      setYears,
      setFilterYear,
      setOnlyMovies
    );
  };

  const callHandleToggleChip = chip => {
    handleToggleChip(chip, filterGenre, setFilterGenre, setShowFilters);
  };

  const callHandleOrder = event => {
    handleOrder(event, setFilterOrder);
  };

  const toggleFilter = () => {
    setShowFilters(!showFilters);
  };

  const toggleGrid = () => {
    setShowGrid(!showGrid);
  };

  let mContent = content;

  return (
    <div>
      <Header
        {...{ showFilters, search, autoComplete, showGrid }}
        handleChange={callHandleChange}
        goTop={goTop}
        toggleFilter={toggleFilter}
        toggleGrid={toggleGrid}
      />
      <FilterBar
        filterResults={filterResults}
        filterYear={filterYear}
        filterGenre={filterGenre}
        filterOrder={filterOrder}
        years={years}
        showFilters={showFilters}
        handleToggleChip={callHandleToggleChip}
        handleFilter={callHandleFilter}
        handleYear={callHandleYear}
        handleOrder={callHandleOrder}
      />
      <main className={classes.content}>
        {mContent.length ? (
          <Grid container spacing={1}>
            {filterOrder === 'Ascending' &&
              //sort alphabetically ascending
              mContent
                .sort((a, b) => {
                  let titleA = a.title.toLowerCase(),
                    titleB = b.title.toLowerCase();
                  if (titleA < titleB) return -1;
                  if (titleA > titleB) return 1;
                  return 0;
                })
                .map(() => {
                  return '';
                })}

            {filterOrder === 'Descending' &&
              //sort alphabetically descending
              mContent
                .sort((a, b) => {
                  let titleA = a.title.toLowerCase(),
                    titleB = b.title.toLowerCase();
                  if (titleA > titleB) return -1;
                  if (titleA < titleB) return 1;
                  return 0;
                })
                .map(() => {
                  return '';
                })}

            {mContent.map((item, index) => {
              if (filterResults === 'movies' && item.type !== 'movie')
                return null;
              if (filterResults === 'tv' && item.type !== 'tv_show')
                return null;
              if (
                filterGenre.on.length !== 0 &&
                !item.genre.some(genre => filterGenre.on.indexOf(genre) !== -1)
              )
                return null;

              if (showGrid) {
                if (item.content) {
                  return item.content.map((child, idx) => (
                    <Grid key={index * idx} item xs={12} sm={6} md={4}>
                      <Tile {...{ ...item, ...child }} />
                    </Grid>
                  ));
                }

                return (
                  <Grid key={index} item xs={12} sm={6} md={4}>
                    <Tile {...item} />
                  </Grid>
                );
              }

              return (
                <Grid key={index} item xs={12}>
                  {item.content ? <MultiCard {...item} /> : <Card {...item} />}
                </Grid>
              );
            })}
          </Grid>
        ) : search ? (
          <div className={classes.error}>
            <ErrorIcon fontSize="large" />
            <p className={testing ? 'test-no-result' : null}>
              No search result.
            </p>
          </div>
        ) : (
          <div className={classes.error}>
            <ErrorIcon fontSize="large" />
            <p className={testing ? 'test-no-data' : null}>
              Can't load the data.
            </p>
          </div>
        )}
      </main>
      <Footer />
    </div>
  );
};
export default withStyles(styles)(App);
