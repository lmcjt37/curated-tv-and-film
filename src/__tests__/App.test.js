import App from '../App';
import Header from '../components/header.js';
import Footer from '../components/footer.js';
import FilterBar from '../components/filterBar.js';
import Card from '../components/card.js';
import MultiCard from '../components/multiCard';
import ErrorIcon from '@material-ui/icons/Error';
import { animateScroll as scroll } from 'react-scroll';
import {
  handleYear,
  handleChange,
  handleFilter,
  handleToggleChip,
  handleOrder,
  toggleFilter
} from '../App';
import content from '../content';

it('render all initial child components', () => {
  const wrapper = mount(<App />);

  expect(wrapper.find(Header)).toHaveLength(1);
  expect(wrapper.find(FilterBar)).toHaveLength(1);
  expect(wrapper.find('main')).toHaveLength(1);
  expect(wrapper.find(Footer)).toHaveLength(1);
  expect(wrapper.find(Card).length).toBeGreaterThan(0);
  expect(wrapper.find(MultiCard).length).toBeGreaterThan(0);
  expect(wrapper.find(ErrorIcon)).toHaveLength(0);
});

it('renders error component for no search results', () => {
  const wrapper = mount(<App testing={true} />);

  wrapper.find('input[type="text"]').simulate('change', {
    target: {
      value: 'random text with no search result'
    }
  });

  expect(wrapper.find(ErrorIcon)).toHaveLength(1);
  expect(wrapper.find('.test-no-result').text()).toEqual('No search result.');
});

it("renders error component when it can't load the data", () => {
  const wrapper = mount(<App testing={true} testType="no data" />);

  expect(wrapper.find(ErrorIcon)).toHaveLength(1);

  expect(wrapper.find('.test-no-data').text()).toEqual("Can't load the data.");
});

it('calls handleChange correctly', () => {
  const data = content;
  let setAutoCompleteArg;
  let setSearchArg;
  let setContentArg;
  const mockSetAutoComplete = arg => (setAutoCompleteArg = arg);
  const mockSetSearch = arg => (setSearchArg = arg);
  const mockSetContent = arg => (setContentArg = arg);

  const event = { target: { value: '' } };
  handleChange(event, data, mockSetAutoComplete, mockSetSearch, mockSetContent);

  expect(setSearchArg).toBeFalsy();
  expect(setAutoCompleteArg).toEqual([]);
  expect(setContentArg.length).toBeGreaterThan(1);

  event.target.value = 'neverToEqualATitle123';
  handleChange(event, data, mockSetAutoComplete, mockSetSearch, mockSetContent);

  expect(setSearchArg).toBeTruthy();
  expect(setAutoCompleteArg).toEqual([]);
  expect(setContentArg.length).toEqual(0);
});

it('calls handleYear correctly', () => {
  let onlyMovies = content.filter(value => value.type === 'movie');
  let setContentArg;
  let setFilterArg;
  const mockSetContent = arg => (setContentArg = arg.length);
  const mockSetFilterYear = arg => (setFilterArg = arg);

  const event = { target: { value: 'All' } };
  handleYear(event, onlyMovies, mockSetFilterYear, mockSetContent);
  let allContentLength = setContentArg;

  expect(allContentLength).toBeGreaterThan(1);

  event.target.value = '2003';
  handleYear(event, onlyMovies, mockSetFilterYear, mockSetContent);

  expect(setContentArg).toBeLessThan(allContentLength);
  expect(setFilterArg).toEqual(event.target.value);
});

it('calls handleFilter correctly for TV', () => {
  let data = content;
  let setFilterResultsArg;
  let setContentArg;
  let setYearsArg;
  let setFilterYearArg;
  const mockSetFilterResults = arg => (setFilterResultsArg = arg);
  const mockSetContent = arg => (setContentArg = arg);
  const mockSetYears = arg => (setYearsArg = arg);
  const mockSetFilterYear = arg => (setFilterYearArg = arg);
  const mockSetOnlyMovies = jest.fn();

  const event = { target: { value: 'all' } };
  handleFilter(
    event,
    data,
    content,
    mockSetFilterResults,
    mockSetContent,
    mockSetYears,
    mockSetFilterYear,
    mockSetOnlyMovies
  );

  expect(setContentArg.length).toBeGreaterThan(1);
  expect(setFilterResultsArg).toEqual('all');
  expect(setFilterYearArg).toEqual('All');
  expect(setYearsArg.length).toBeGreaterThan(0);

  event.target.value = 'tv';
  handleFilter(
    event,
    data,
    content,
    mockSetFilterResults,
    mockSetContent,
    mockSetYears,
    mockSetFilterYear,
    mockSetOnlyMovies
  );

  expect(setFilterResultsArg).toEqual('tv');
  expect(setYearsArg).toEqual([]);
  expect(setFilterYearArg).toEqual('All');
  expect(mockSetOnlyMovies).toHaveBeenCalledTimes(1);
});

it('calls handleFilter correctly for Movies', () => {
  let data = content;
  let setFilterResultsArg;
  let setContentArg;
  let setYearsArg;
  let setFilterYearArg;
  const mockSetFilterResults = arg => (setFilterResultsArg = arg);
  const mockSetContent = arg => (setContentArg = arg);
  const mockSetYears = arg => (setYearsArg = arg);
  const mockSetFilterYear = arg => (setFilterYearArg = arg);
  const mockSetOnlyMovies = jest.fn();

  const event = { target: { value: 'movies' } };
  handleFilter(
    event,
    data,
    content,
    mockSetFilterResults,
    mockSetContent,
    mockSetYears,
    mockSetFilterYear,
    mockSetOnlyMovies
  );

  expect(setContentArg.length).toBeGreaterThan(1);
  expect(setFilterResultsArg).toEqual('movies');
  expect(setYearsArg).toEqual([
    2019,
    2017,
    2016,
    2014,
    2013,
    2007,
    2005,
    2004,
    2003,
    2001,
    1999,
    1998,
    1995,
    1994,
    1993,
    1972
  ]);
  expect(setFilterYearArg).toEqual('All');
  expect(mockSetOnlyMovies).toHaveBeenCalledTimes(1);
});

it('calls handleToggleChip correctly', () => {
  const filterGenre = {
    available: [
      'Action',
      'Adventure',
      'Animation',
      'Biography',
      'Comedy',
      'Crime',
      'Drama',
      'Family',
      'Fantasy',
      'History',
      'Horror',
      'Mystery',
      'Romance',
      'Sci-Fi',
      'Thriller'
    ],
    on: []
  };
  let setFilterGenreArg;
  let setShowFiltersArg;
  const mockSetFilterGenre = arg => (setFilterGenreArg = arg);
  const mockSetShowFilters = arg => (setShowFiltersArg = arg);

  let chip = 'Action';
  handleToggleChip(chip, filterGenre, mockSetFilterGenre, mockSetShowFilters);

  expect(setFilterGenreArg).toEqual({
    available: [
      'Action',
      'Adventure',
      'Animation',
      'Biography',
      'Comedy',
      'Crime',
      'Drama',
      'Family',
      'Fantasy',
      'History',
      'Horror',
      'Mystery',
      'Romance',
      'Sci-Fi',
      'Thriller'
    ],
    on: ['Action']
  });

  chip = 'Action';
  handleToggleChip(chip, filterGenre, mockSetFilterGenre, mockSetShowFilters);

  expect(setFilterGenreArg).toEqual({
    available: [
      'Action',
      'Adventure',
      'Animation',
      'Biography',
      'Comedy',
      'Crime',
      'Drama',
      'Family',
      'Fantasy',
      'History',
      'Horror',
      'Mystery',
      'Romance',
      'Sci-Fi',
      'Thriller'
    ],
    on: []
  });
  expect(setShowFiltersArg).toEqual(true);
});

it('calls handleOrder correctly', () => {
  let setFilterOrderArg;
  const mockSetFilterOrder = arg => (setFilterOrderArg = arg);

  const event = { target: { value: 'Ascending' } };
  handleOrder(event, mockSetFilterOrder);
  expect(setFilterOrderArg).toEqual('Ascending');

  event.target.value = 'Descending';
  handleOrder(event, mockSetFilterOrder);
  expect(setFilterOrderArg).toEqual('Descending');
});

it('calls toggleFilter correctly', () => {
  let setShowFiltersArg;
  const setShowFilters = arg => (setShowFiltersArg = arg);

  let showFilters = true;
  toggleFilter(setShowFilters, showFilters);
  expect(setShowFiltersArg).toBeFalsy();

  showFilters = false;
  toggleFilter(setShowFilters, showFilters);
  expect(setShowFiltersArg).toBeTruthy();
});

xit('calls goTop correctly', () => {
  scroll.scrollToTop = jest.fn();

  const wrapper = shallow(<App />).dive();

  wrapper.instance().goTop();

  expect(scroll.scrollToTop).toHaveBeenCalledTimes(1);
});

it('snapshot changes when input change', () => {
  const component = render(<App />);
  expect(component).toMatchSnapshot();
});
