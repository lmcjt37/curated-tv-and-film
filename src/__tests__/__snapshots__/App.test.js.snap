// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot changes when input change 1`] = `[Function]`;
